cmd/registry/main.go

```diff
@@ -68,6 +70,30 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Start profiling if enabled.
+	if *profiling {
+		// Optional trace agent URL override.
+		traceAgentURL := "localhost:8126"
+		if e := os.Getenv("TRACE_AGENT_URL"); e != "" {
+			traceAgentURL = e
+		}
+
+		err = profiler.Start(
+			profiler.WithService("kafka-registry"),
+			profiler.WithAgentAddr(traceAgentURL),
+			profiler.WithProfileTypes(
+				profiler.CPUProfile,
+				profiler.HeapProfile,
+			),
+		)
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	}
+
+	defer profiler.Stop()
```

Comment:
nit, should be inside the if

Reply:
Oh yes, just added the condition and definitely didn't catch this